### PR TITLE
operator: deprecation announcement of fetching charts

### PR DESCRIPTION
**Pull request by zirain**
**Please provide a description of this PR:**

part of https://github.com/istio/istio/issues/41704

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [x] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: releasenotes/notes/deprecate-operator-httpfetch.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: release-notes/v2
+kind: feature
+area: installation
+releaseNotes:
+- |
+  **Added** deprecation announcement of fetching charts from arbitrary URLs in `istio-operator`.
```
Comment (ericvn) [inline releasenotes/notes/deprecate-operator-httpfetch.yaml:6]: Deprecated is an acceptable verb, so maybe:
```suggestion
  **Deprecated** fetching charts from arbitrary URLs in `istio-operator`.
```
The deprecation gets lost within the text. This should better highlight it.

File: operator/pkg/manifest/shared.go [hunk 2]
```diff
@@ -383,6 +383,8 @@ func RewriteURLToLocalInstallPath(installPackagePath, profileOrPath string, skip
 		return "", "", err
 	}
 	if isURL {
+		// TODO: remove this feature in next major release
+		log.Warnf("fetching charts from arbitrary URLs will be removed in next major released")
```
Comment (dhawton) [inline operator/pkg/manifest/shared.go:387]: Nit:

```Fetching charts from arbitrary URLs will be removed in the next major release```

Comment (litong01) [reply operator/pkg/manifest/shared.go:387]: @zirain I am not exactly sure what arbitrary URLs means, from looking at the code, it seems to me that any URL will not be supported. Do we have `arbitrary` URL vs `non arbitrary` or `official` URLs? If not, I would think this is basically saying we no longer support using URLs?

**Review by litong01 — CHANGES_REQUESTED** [review]

File: operator/pkg/manifest/shared.go (hunk 2 above)
Comment (howardjohn) [reply operator/pkg/manifest/shared.go:387]: The intent is no URLs. Users should use local files. Fetching arbitrary tarballs from the internet and then applying them as root to the cluster is problematic on many levels, and not actually a use case istio users really have; they apply local files or built in manifests

Comment (litong01) [reply operator/pkg/manifest/shared.go:387]: @zirain @howardjohn right, in that case, we should simply say URL pointed charts wont be supported or deprecated in the future. `arbitrary` just seems to me not very clear.

Comment (zirain) [reply operator/pkg/manifest/shared.go]: remove `arbitrary` from log

File: releasenotes/notes/deprecate-operator-httpfetch.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+apiVersion: release-notes/v2
+kind: feature
+area: installation
+releaseNotes:
+- |
+  **Deprecated** fetching charts from arbitrary URLs in `istio-operator`.
```
Comment (ericvn) [inline releasenotes/notes/deprecate-operator-httpfetch.yaml:6]: Should we remove arbitrary here as well?

Comment (zirain) [reply releasenotes/notes/deprecate-operator-httpfetch.yaml:6]: sounds good

**Review by dhawton — APPROVED** [review]

**Review by litong01 — APPROVED** [review]
Thanks very much for your contributions.

**Pull request merged**